elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/HibernateTransaction.java

```diff
@@ -393,37 +351,39 @@ public T getObject() {
     }
 
-    @Override
-    public <T> Object getRelation(
+    public Object getRelation(
+            DataStoreTransaction relationTx,
             Object entity,
-            RelationshipType relationshipType,
             String relationName,
-            Class<T> relationClass,
-            EntityDictionary dictionary,
             Optional<FilterExpression> filterExpression,
-            Sorting sorting,
-            Pagination pagination
-    ) {
-        Object val = com.yahoo.elide.core.PersistentResource.getValue(entity, relationName, dictionary);
-        if (val instanceof Collection) {
-            Collection filteredVal = (Collection) val;
-            if (filteredVal instanceof AbstractPersistentCollection) {
-                Optional<Sorting> sortingRules = sorting != null ? Optional.of(sorting) : Optional.empty();
-                Optional<Pagination> paginationRules = pagination != null ? Optional.of(pagination) : Optional.empty();
-
-                @SuppressWarnings("unchecked")
-                final Optional<Query> possibleQuery = new HQLTransaction.Builder<>(session, filteredVal, relationClass,
-                        dictionary)
-                        .withPossibleFilterExpression(filterExpression)
-                        .withPossibleSorting(sortingRules)
-                        .withPossiblePagination(paginationRules)
-                        .build();
-                if (possibleQuery.isPresent()) {
-                    return possibleQuery.get().list();
+            Optional<Sorting> sorting,
+            Optional<Pagination> pagination,
+            RequestScope scope) {
+        try {
+            com.yahoo.elide.core.RequestScope requestScope = (com.yahoo.elide.core.RequestScope) scope;
+            EntityDictionary dictionary = requestScope.getDictionary();
+            Object val = com.yahoo.elide.core.PersistentResource.getValue(entity, relationName, dictionary);
+            if (val instanceof Collection) {
+                Collection filteredVal = (Collection) val;
+                if (filteredVal instanceof AbstractPersistentCollection) {
+                    @SuppressWarnings("unchecked")
+                    Class<?> relationClass = dictionary.getParameterizedType(entity, relationName);
+                    final Optional<Query> possibleQuery =
+                            new HQLTransaction.Builder<>(session, filteredVal, relationClass,
+                            dictionary)
+                            .withPossibleFilterExpression(filterExpression)
+                            .withPossibleSorting(sorting)
+                            .withPossiblePagination(pagination)
+                            .build();
+                    if (possibleQuery.isPresent()) {
+                        return possibleQuery.get().list();
+                    }
                 }
             }
+            return val;
+        } catch (ClassCastException e) {
```

Comment:
Keep this catch block closer to the cast.  The only statement in the try should be the cast.